File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution
{
public:
    bool canPlaceFlowers(vector<int> &flowerbed, int n)
    {
        int s = flowerbed.size();
        for (int i = 0; i < s; ++i)
        {
            if (flowerbed[i] == 0)
            {
                if (i == 0 && (s == 1 || flowerbed[i + 1] == 0))
                {
                    --n;
                    flowerbed[i] = 1;
                }
                if (i == (s - 1) && (s == 1 || flowerbed[i - 1] == 0))
                {
                    --n;
                    flowerbed[i] = 1;
                }
                if ((i - 1) > 0 && flowerbed[i - 1] == 0 && (i + 1) < s && flowerbed[i + 1] == 0)
                {
                    --n;
                    flowerbed[i] = 1;
                }
            }
        }

        return n <= 0;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0:
            return True

        l = len(flowerbed)
        for i in range(l):
            if flowerbed[i] == 1:
                continue

            if i - 1 >= 0 and flowerbed[i - 1] == 1:
                continue

            if i + 1 < l and flowerbed[i + 1] == 1:
                continue

            flowerbed[i] = 1
            n -= 1

            if n == 0:
                return True

        return False